src/calldata/evm.ts

```diff
@@ -0,0 +1,366 @@
+import { Buffer } from 'buffer/';
+import { keccak256 } from 'js-sha3';
+import { encode } from 'rlp';
+
+/**
+* Look through an ABI definition to see if there is a function 
+* that matches the signature provided.
+* @param sig    a 0x-prefixed hex string containing 4 bytes of info
+* @param abi    a Solidity JSON ABI structure
+*               (https://docs.ethers.io/v5/api/utils/abi/formats/#abi-formats--solidity)
+* @returns      Buffer containing RLP-serialized array of 
+*               calldata info to pass to signing request
+*/ 
+export const parseSolidityJSONABI = function(sig: string, abi: any[]): Buffer {
+  sig = coerceSig(sig);
+  let match = null;
+  abi.forEach((item) => {
```

Comment:
This will set `match` to the last matching def. I'm not sure if we want that behavior, or if there's only one that should match. 

If we want the first (or only), we could use `Array.find()` like this:

```
  const match = abi.find((item) => {
    if (item.type === 'function') {
      const def = parseDef(item);
      // If this matches the function selector (sig) we can return it
      return (def && getFuncSig(def.canonicalName) === sig) 
    }
  })
  
  if (match && match.def) {
    return Buffer.from(encode(match));
  }
  return null;
```

Reply:
The probability of any two functions having the same sig is 1/2**32 so not worried about it, but I like your syntax better anyway :)

Reply:
[756bc00](https://github.com/GridPlus/gridplus-sdk/pull/353/commits/756bc001e62e4ba9fce75734645727253137ddc2)